# kaleidofin
##  Task : 
	•	What is the Annualized  Return of the person who bids in the last month ? 
	•	What is the Annualized Return of the person who bids in the first month ?
	

### Importing Data and Pandas library

In [31]:
import pandas as pd
data = pd.read_csv("/home/prasad/Desktop/Assignment_1_2/Assignment_1 2/chit fund exercise.csv")
data.head()

Month  Contribution  Amount won by the bidder  \
0      1          2000                     40000   
1      2          2000                     42000   
2      3          2000                     45000   
3      4          2000                     48000   
4      5          2000                     40000   

   Chit fund organizer commission  Net amount recd by Bid winner  \
0                            2500                          37500   
1                            2500                          39500   
2                            2500                          42500   
3                            2500                          45500   
4                            2500                          37500   

   Amount returned to everyone in the group  
0                                       400  
1                                       320  
2                                       200  
3                                        80  
4                                       400

In [32]:
data.columns

Index(['Month', 'Contribution', 'Amount won by the bidder',
       'Chit fund organizer commission', 'Net amount recd by Bid winner',
       'Amount returned to everyone in the group'],
      dtype='object')

### Calculating Actual Contribution contributed by each member of the chit fund for every month.

In [33]:
data["Actual Contribution"] = data["Contribution"] - data["Amount returned to everyone in the group"]
data.head()

Month  Contribution  Amount won by the bidder  \
0      1          2000                     40000   
1      2          2000                     42000   
2      3          2000                     45000   
3      4          2000                     48000   
4      5          2000                     40000   

   Chit fund organizer commission  Net amount recd by Bid winner  \
0                            2500                          37500   
1                            2500                          39500   
2                            2500                          42500   
3                            2500                          45500   
4                            2500                          37500   

   Amount returned to everyone in the group  Actual Contribution  
0                                       400                 1600  
1                                       320                 1680  
2                                       200                 1800  
3                                        80                 1920  
4                                       400                 1600

### Calculating Total Contribution contributed by each member of the chit fund throughout 25 months. <br>
### Total amount contributed by each member = Sum of (Contribution of every month(2000) - Amount returned) for 25 months

In [34]:
total = data["Actual Contribution"].sum()
total

43800

### Calculating Return/Net Profit obtained by the Bid Winners for every month <br>
### Return =  Net amount received by a participant - total amount contributed by a participant

In [35]:
data["return"] = data["Net amount recd by Bid winner"] - total
data.head()

Month  Contribution  Amount won by the bidder  \
0      1          2000                     40000   
1      2          2000                     42000   
2      3          2000                     45000   
3      4          2000                     48000   
4      5          2000                     40000   

   Chit fund organizer commission  Net amount recd by Bid winner  \
0                            2500                          37500   
1                            2500                          39500   
2                            2500                          42500   
3                            2500                          45500   
4                            2500                          37500   

   Amount returned to everyone in the group  Actual Contribution  return  
0                                       400                 1600   -6300  
1                                       320                 1680   -4300  
2                                       200                 1800   -1300  
3                                        80                 1920    1700  
4                                       400                 1600   -6300

### Calculating Return Percentage obtained by the Bid Winners for every month <br>
### Return percentage = (Return / Total amount invested) X 100

In [36]:
data["ret%"] = (data["return"]/total)*100
data.head()

Month  Contribution  Amount won by the bidder  \
0      1          2000                     40000   
1      2          2000                     42000   
2      3          2000                     45000   
3      4          2000                     48000   
4      5          2000                     40000   

   Chit fund organizer commission  Net amount recd by Bid winner  \
0                            2500                          37500   
1                            2500                          39500   
2                            2500                          42500   
3                            2500                          45500   
4                            2500                          37500   

   Amount returned to everyone in the group  Actual Contribution  return  \
0                                       400                 1600   -6300   
1                                       320                 1680   -4300   
2                                       200                 1800   -1300   
3                                        80                 1920    1700   
4                                       400                 1600   -6300   

        ret%  
0 -14.383562  
1  -9.817352  
2  -2.968037  
3   3.881279  
4 -14.383562

# Answers

## Q1. What is the Annualized  Return of the person who bids in the last month ? 

## Formula for Annualized Return which I was able to procure after reading on internet:

In [37]:
data.iloc[24]

Month                                          25.000000
Contribution                                 2000.000000
Amount won by the bidder                    50000.000000
Chit fund organizer commission               2500.000000
Net amount recd by Bid winner               47500.000000
Amount returned to everyone in the group        0.000000
Actual Contribution                          2000.000000
return                                       3700.000000
ret%                                            8.447489
Name: 24, dtype: float64

In [38]:
((((1 + (data["ret%"][24])/100)) ** (12/25)) - 1) * 100

3.969357358648673

## Annualized return of the person who bids in last month is: 3.97%

## Q2. What is the Annualized  Return of the person who bids in the first month ? 

In [39]:
data.iloc[0]

Month                                           1.000000
Contribution                                 2000.000000
Amount won by the bidder                    40000.000000
Chit fund organizer commission               2500.000000
Net amount recd by Bid winner               37500.000000
Amount returned to everyone in the group      400.000000
Actual Contribution                          1600.000000
return                                      -6300.000000
ret%                                          -14.383562
Name: 0, dtype: float64

In [40]:
((((1 + (data["ret%"][0])/100)) ** (12/25)) - 1) * 100

-7.183019602665885

## Annualized return of the person who bids in first month is: -7.18%

## Q3. Write an Python script which calculates the annualized return of chit fund participant?      - Show the Return % for each month's bid winner.



## Return percentages of each month's bid winners.

In [41]:
data["ret%"]

0    -14.383562
1     -9.817352
2     -2.968037
3      3.881279
4    -14.383562
5     -9.817352
6     -7.534247
7     -5.251142
8    -14.383562
9     -9.817352
10    -2.968037
11     3.881279
12   -14.383562
13    -9.817352
14    -7.534247
15    -7.534247
16   -12.100457
17    -9.817352
18    -2.968037
19    -2.968037
20    -5.251142
21    -0.684932
22     1.598174
23     3.881279
24     8.447489
Name: ret%, dtype: float64